src/shopilingus/Shop.java

```diff
@@ -0,0 +1,139 @@
+package shopilingus;
+
+import java.util.LinkedList;
+import java.util.List;
+
+public class Shop {
+    private static int nextId = 0;
+
+    public int id;
+    public String name;
+    private Location location;
+
+    private List<String> types;
+    private List<ProductDetails> products;
+    public List<Service> services;
+    private List<Employee> employees;
+
+
+    public Shop(String name, int floor, int box) {
+        this.id = Shop.nextId++;
+        this.name = name;
+        this.location = new Location(floor, box);
+        this.types = types;
+        this.products = new LinkedList<ProductDetails>();
+        this.services = new LinkedList<Service>();
+        this.employees = new LinkedList<Employee>();
+    }
+
+    public static int getNextId() {
```

Comment:
please remove getter and setter for this static variable. encapsulation - we want to keep `nextId` fully private to don't let anybody hurt us by changing it ;)